registry/auth/htpasswd/access.go

```diff
@@ -111,6 +119,34 @@ func (ch challenge) Error() string {
 	return fmt.Sprintf("basic authentication challenge for realm %q: %s", ch.realm, ch.err)
 }
 
+// createHtpasswdFile creates and populates htpasswd file with a new user in case the file is missing
+func createHtpasswdFile(path string) error {
+	if _, err := os.Open(path); os.IsNotExist(err) {
```

Comment:
Put the bulk of the work in the main path, not the indented path.

Reply:
Any other non-empty error should also be returned

Reply:
Thanks @stevvooe @dmcgowan fixed.